core/src/main/scala/filodb.core/columnstore/Analyzer.scala

```diff
@@ -0,0 +1,92 @@
+  }
+
+  def prettify(name: String, maxBarLen: Int = 60): String = {
+    val head = s"===== $name =====\n  Min: $min\n  Max: $max\n  Average: ${sum/numElems.toDouble} ($numElems)"
+    val maxQuantity = Try(buckets.values.max).getOrElse(0)
+    buckets.toSeq.sortBy(_._1).foldLeft(head + "\n") { case (str, (bucket, num)) =>
+      val numChars = (num * maxBarLen) / maxQuantity
+      str + "| %08d: %08d %s".format(bucket, num, "X" * numChars) + "\n"
+    }
+  }
+}
+
+object Histogram {
+  val empty: Histogram = Histogram(Int.MaxValue, Int.MinValue, 0, 0, Map.empty)
+}
+
+case class ColumnStoreAnalysis(numSegments: Int,
+                               numPartitions: Int,
+                               rowsInSegment: Histogram,
+                               chunksInSegment: Histogram,
+                               segmentsInPartition: Histogram) {
+  def prettify(): String = {
+    s"ColumnStoreAnalysis\n  numSegments: $numSegments\n  numPartitions: $numPartitions\n" +
+    rowsInSegment.prettify("# Rows in a segment") +
+    chunksInSegment.prettify("# Chunks in a segment") +
+    segmentsInPartition.prettify("# Segments in a partition")
+  }
+}
+
+/**
+ * Analyzes the segments and chunks for a given dataset/version.  Gives useful information
+ * about distribution of segments and chunks within segments.  Should be run offline, as could take a while.
+ */
+object Analyzer {
+  val NumSegmentsBucketKeys = Array(0, 10, 50, 100, 500)
+  val NumChunksPerSegmentBucketKeys = Array(0, 5, 10, 25, 50, 100)
+  val NumRowsPerSegmentBucketKeys = Array(0, 10, 100, 1000, 5000, 10000, 50000)
+
+  def analyze(cs: CachedMergingColumnStore, dataset: TableName, version: Int): ColumnStoreAnalysis = {
+    var numSegments = 0
+    var rowsInSegment: Histogram = Histogram.empty
+    var chunksInSegment: Histogram = Histogram.empty
+    var segmentsInPartition: Histogram = Histogram.empty
+    val partitionSegments = (new collection.mutable.HashMap[PartitionKey, Int]).withDefaultValue(0)
+
+    for { param <- cs.getScanSplits(dataset)
+          rowmaps = Await.result(cs.scanChunkRowMaps(dataset, version, (p: PartitionKey) => true,
+                                                     params = param), 5.minutes)
+          (partKey, _, rowmap) <- rowmaps } yield {
+      // Figure out # chunks and rows per segment
+      val numRows = rowmap.chunkIds.length
+      val numChunks = rowmap.nextChunkId
+
+      numSegments = numSegments + 1
+      rowsInSegment = rowsInSegment.add(numRows, NumRowsPerSegmentBucketKeys)
+      chunksInSegment = chunksInSegment.add(numChunks, NumChunksPerSegmentBucketKeys)
+      partitionSegments(partKey) += 1
+    }
+
+    for { (partKey, numSegments) <- partitionSegments } {
+      segmentsInPartition = segmentsInPartition.add(numSegments, NumSegmentsBucketKeys)
+    }
+
+    ColumnStoreAnalysis(numSegments, partitionSegments.size,
+                        rowsInSegment, chunksInSegment, segmentsInPartition)
+  }
+}
```

Comment:
Would it make more sense, since this creates a ColumnStoreAnalysis, that it is an apply?
     object ColumnStoreAnalysis  {
 
         def apply(cs: CachedMergingColumnStore, dataset: TableName, version: Int): ColumnStoreAnalysis = {
            var numSegments = 0
            var rowsInSegment: Histogram = Histogram.empty
            var chunksInSegment: Histogram = Histogram.empty
            var segmentsInPartition: Histogram = Histogram.empty
            val partitionSegments = (new collection.mutable.HashMap[PartitionKey, Int]).withDefaultValue(0)

           for { 
              param                        <- cs.getScanSplits(dataset)
              rowmaps                    = Await.result(cs.scanChunkRowMaps(dataset, version, (p: PartitionKey) => true, 
                                                        params = param), 5.minutes)
              (partKey, _, rowmap) <- rowmaps 
           } yield {
               // Figure out # chunks and rows per segment
               val numRows = rowmap.chunkIds.length
               val numChunks = rowmap.nextChunkId

               numSegments = numSegments + 1
               rowsInSegment = rowsInSegment.add(numRows, NumRowsPerSegmentBucketKeys)
               chunksInSegment = chunksInSegment.add(numChunks, NumChunksPerSegmentBucketKeys)
               partitionSegments(partKey) += 1
          }

          for { (partKey, numSegments) <- partitionSegments } {
              segmentsInPartition = segmentsInPartition.add(numSegments, NumSegmentsBucketKeys)
          }

          ColumnStoreAnalysis(numSegments, partitionSegments.size,
                rowsInSegment, chunksInSegment, segmentsInPartition)
        }
     }